zingo-memo/src/lib.rs

```diff
@@ -65,34 +65,36 @@ pub fn parse_zingo_memo(memo: [u8; 511]) -> io::Result<ParsedMemo> {
 /// A helper function to encode a UA as a CompactSize specifying the number
 /// of receivers, followed by the UA's raw encoding as specified in
 /// <https://zips.z.cash/zip-0316#encoding-of-unified-addresses>
-pub fn write_unified_address_to_raw_encoding<W: Write>(
+pub fn write_unified_address_to_raw_encoding<W: Write + Clone>(
     ua: &UnifiedAddress,
     writer: W,
 ) -> io::Result<()> {
     let mainnet_encoded_ua = ua.encode(&zcash_primitives::consensus::MAIN_NETWORK);
     let (_mainnet, address) = Address::decode(&mainnet_encoded_ua).unwrap();
     let receivers = address.items();
-    Vector::write(writer, &receivers, |mut w, receiver| {
+    Vector::write(writer, &receivers, |w, receiver| {
         let (typecode, data): (u32, &[u8]) = match receiver {
             Receiver::Orchard(ref data) => (3, data),
             Receiver::Sapling(ref data) => (2, data),
             Receiver::P2pkh(ref data) => (0, data),
             Receiver::P2sh(ref data) => (1, data),
             Receiver::Unknown { typecode, ref data } => (*typecode, data.as_slice()),
         };
-        CompactSize::write(&mut w, typecode as usize)?;
-        CompactSize::write(&mut w, data.len())?;
+        CompactSize::write(w.clone(), typecode as usize)?;
```

Comment:
i will need to do some research on these rust changes but cloning readers / writers doesnt sit right with me....